app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,25 @@
+class SessionsController < ApplicationController
+  
+  before_action only: :sign_in do |c| 
+    c.available_to "logged_out"
+  end
+
+  def sign_in
+  end
+
+  def logout
+    session[:user_id] = nil
+    redirect_to root_path
+  end
+
+  def checking_credentials
```

Comment:
Generally you would post to /sign_in as well, that way the url will not change when you `render :sign_in`. Currently the url changes to /checking_credentials, which is a bit funky.